dpppt-backend-sdk/dpppt-backend-sdk-data/src/main/java/org/dpppt/backend/sdk/data/gaen/JDBCGAENDataServiceImpl.java

```diff
@@ -99,6 +99,11 @@ public int getMaxExposedIdForKeyDate(
             + " and rolling_start_number < :rollingPeriodStartNumberEnd"
             + " and received_at < :publishedUntil";
 
+    params.addValue(
+        "maxAllowedStartNumber",
+        now.roundToPreviousBucket(releaseBucketDuration).plusHours(2).get10MinutesSince1970());
+    sql += " and rolling_start_number < :maxAllowedStartNumber";
```

Comment:
@UBaggeler should this be the same as in `getMaxExposedIdForKeyDate`?

```suggestion
    sql += " and rolling_start_number + rolling_perdiod < :maxAllowedStartNumber";
```

Reply:
So this should be PLUS TWO HOUS, since according to the [Google and Apple specification](https://developer.apple.com/documentation/exposurenotification/setting_up_a_key_server?language=objc), due to time skew, Google and Apple will still count exposures with keys which have expired for up to two hours. 
> Must not distribute temporary exposure key data until at least 2 hours after the end of the keyʼs expiration window

We definitely should add comments though and put it into a constant!

Reply:
haha well the plus should be on the left side, so minus on the right side....

@ineiti @martinalig  please confirm and I'll change it!

Reply:
yes, now it makes sense. Double-negative, got me...

Intuitive/qualitative reasoning to confirm minus: with minus, `maxAllowedStartNumber` gets smaller, so less keys are released. With plus, more keys would be released, which must be wrong if we need to keep them.